src/v/cluster/topics_frontend.cc

```diff
@@ -97,6 +100,25 @@ needs_linearizable_barrier(const std::vector<topic_result>& results) {
 ss::future<std::vector<topic_result>> topics_frontend::create_topics(
   std::vector<custom_assignable_topic_configuration> topics,
   model::timeout_clock::time_point timeout) {
+    for (auto& tp : topics) {
+        if (!tp.cfg.properties.shadow_indexing.has_value()) {
+            tp.cfg.properties.shadow_indexing
+              = _metadata_cache.get_default_shadow_indexing_mode();
+        }
```

Comment:
This seems like a lot of code change for a check that needs to happen in the CreateTopics API. Are there other APIs or places that had the same problem as reported in https://github.com/redpanda-data/redpanda/issues/13492?

Reply:
>seems like a lot of code change

I wouldn't really dispute that. General idea here is that any topic should reflect certain global defaults regardless of how it was created (auto vs explicit), so most of the code motion is to avoid duplicating `if read && write ... else if read ... else if write` yet again. There might be another way to skin that cat.

>check that needs to happen in the CreateTopics API

You mean on the `kafka` side?

Reply:
@NyaliaLui raises a good point about when 'configuration' validation should be occurring.  My general feel is that the `topic_front` should be the final arbiter (with support from the kafka API front end).

I think it may be 'better' for `topic_frontend` to handle this in the event some 'other' way of creating topics is introduced (e.g. via the admin API).  That would reduce the possibility of missing a validation in an implementation.

Reply:
> You mean on the kafka side?

Yes, that is what I mean. However, considering the explanation so far, I agree that keeping this in the topics frontend is better for future-proofing.